Patterns and offsets for ARM Tahoe Beta 2 added.

tbiehn opened the pull request:
Patterns and offsets for ARM Tahoe Beta 2 added.

tbiehn commented:
Working on Beta 6.

agigao commented:
Merge, please :)

Teeco123 commented:
is focusing spaces not functional? I'm using yabai from your branch on beta 6 and M1 mac and it does not work for me. There is no error with scripting addition but issue is maybe on my side tho

tbiehn commented:
@Teeco123 - yep, works. Take a look at console.app, filter for 'yabai-sa', spam --uninstall-sa // --load-sa, tell me what hooks and what doesn't.
You should see a log in there about Tahoe there as well.

tbiehn commented:
Working on beta 8.

koekeishiya commented:
If all patterns are correct; please update it once the Tahoe RC is out, and I'll merge it then.

You can RE the x86-64 image on an Apple Silicon macbook and find these offsets as well; the patterns/identifiers are usually similar. This is what I do (as I no longer have an intel-based macbook myself).

I can do the x86-64 part when the RC releases as well, if you don't want to.

roelvangils commented:
@tbiehn, I've just tested the Tahoe fork on **macOS 26.0 (BuildVersion: 25A5351b)** and wanted to share my findings to help with tracking compatibility.

## What I did:

1. Successfully built yabai from the Tahoe fork (v7.1.15)
2. Installed to `/usr/local/bin/yabai` with ad-hoc code signing
3. Verified SIP is partially disabled (Filesystem Protections: disabled, as required)
4. Confirmed yabai has Accessibility permissions in TCC database (yep)
5. Load scripting addition via `sudo yabai --load-sa` and by restarting Dock

## What's working
- Window management commands (focus, move, resize)
- Window opacity changes (e.g., `yabai -m window --opacity 0.8` works perfectly!)
- Query commands for windows and spaces
- skhd integration for keyboard shortcuts
- All non-scripting-addition features

## What's NOT working
- Space switching commands (e.g., `yabai -m space --focus 2`)
- Returns error: `cannot focus space due to an error with the scripting-addition`

## Conclusion:
It appears the scripting addition injection into Dock.app isn't compatible with build 25A5351 yet. 

Happy to test any patches or provide additional debugging info if needed!

tbiehn commented:
> @tbiehn, I've just tested the Tahoe fork on **macOS 26.0 (BuildVersion: 25A5351b)** and wanted to share my findings to help with tracking compatibility.
> 
> ## What I did:
> 1. Successfully built yabai from the Tahoe fork (v7.1.15)
> 2. Installed to `/usr/local/bin/yabai` with ad-hoc code signing
> 3. Verified SIP is partially disabled (Filesystem Protections: disabled, as required)
> 4. Confirmed yabai has Accessibility permissions in TCC database (yep)
> 5. Load scripting addition via `sudo yabai --load-sa` and by restarting Dock
> 
> ## What's working
> * Window management commands (focus, move, resize)
> * Window opacity changes (e.g., `yabai -m window --opacity 0.8` works perfectly!)
> * Query commands for windows and spaces
> * skhd integration for keyboard shortcuts
> * All non-scripting-addition features
> 
> ## What's NOT working
> * Space switching commands (e.g., `yabai -m space --focus 2`)
> * Returns error: `cannot focus space due to an error with the scripting-addition`
> 
> ## Conclusion:
> It appears the scripting addition injection into Dock.app isn't compatible with build 25A5351 yet.
> 
> Happy to test any patches or provide additional debugging info if needed!

Thanks for reporting -- you may not be loading the SA. You'll get detailed logs from `console.app`;
<img width="881" height="731" alt="image" src="https://github.com/user-attachments/assets/790c7f62-50c8-4ae8-9e8e-e3ea73d92d9a" />

Verified working on arm for beta 9.

roelvangils commented:
Thanks, I must note that I am on Public (!) Version 26.0 Beta (25A5351b), but the build number is identical to Developer Beta 9, so that can't be the issue, no?

I believe the scripting edition is indeed loaded, 'cause I don't get any errors anymore:

``
yabai -m space --switch 1
cannot focus an already focused space.
``

```
yabai -m space --switch 2
```

(Here it looks like Yabai silently thinks it has switched to space to, but nothing happened.) :) 

Console output:

```
default	11:31:56.694643+0200	yabai	[0x745050640] Connection returned listener port: 0x4e03
default	11:31:58.916399+0200	yabai	[0x10520ddb0] activating connection: mach=true listener=false peer=false name=com.apple.cfprefsd.daemon
default	11:31:58.965888+0200	loader	[0x103590210] activating connection: mach=true listener=false peer=false name=com.apple.coreservices.launchservicesd
default	11:31:58.966412+0200	loader	[0x1035914b0] activating connection: mach=true listener=false peer=false name=com.apple.analyticsd
default	11:31:58.968847+0200	loader	[0x103595b60] activating connection: mach=true listener=false peer=false name=com.apple.cfprefsd.daemon
default	11:31:58.989258+0200	loader	Entering exit handler.
default	11:31:58.989269+0200	loader	Queueing exit procedure onto XPC queue. Any further messages sent will be discarded. activeSendTransactions=0
default	11:31:58.989326+0200	loader	Cancelling XPC connection. Any further reply handler invocations will not retry messages
default	11:31:58.989330+0200	loader	[0x1035914b0] invalidated because the current process cancelled the connection by calling xpc_connection_cancel()
default	11:31:58.990238+0200	yabai	[0x105210b60] activating connection: mach=true listener=false peer=false name=com.apple.distributed_notifications@Uv3
default	11:31:58.989342+0200	loader	Exiting exit handler.
default	11:31:58.990330+0200	yabai	[0x10520f6a0] activating connection: mach=true listener=false peer=false name=com.apple.usernoted.client
default	11:31:58.989350+0200	loader	XPC message reply connection invalidated (client likely exiting): Connection invalid
default	11:31:58.992047+0200	usernoted	Connection com.koekeishiya.yabai with path: /usr/local/bin/yabai
default	11:31:58.991652+0200	yabai	[0x80542c000] activating connection: mach=true listener=false peer=false name=com.apple.lsd.mapdb
default	11:31:58.992685+0200	yabai	[0x80542c140] activating connection: mach=true listener=false peer=false name=com.apple.iconservices
default	11:31:58.993446+0200	yabai	[0x80542c280] activating connection: mach=true listener=false peer=false name=com.apple.iconservices.store
default	11:31:58.994316+0200	yabai	[0x80542c640] activating connection: mach=true listener=false peer=false name=com.apple.tccd.system
default	11:31:58.994649+0200	tccd	AUTHREQ_ATTRIBUTION: msgID=10577.1, attribution={responsible={TCCDProcess: identifier=dev.warp.Warp-Stable, pid=1029, auid=501, euid=501, responsible_path=/Applications/Warp.app/Contents/MacOS/stable, binary_path=/Applications/Warp.app/Contents/MacOS/stable}, requesting={TCCDProcess: identifier=com.koekeishiya.yabai, pid=10577, auid=501, euid=0, binary_path=/usr/local/bin/yabai}, },
default	11:31:59.002962+0200	yabai	[0x80542c640] invalidated after the last release of the connection object
default	11:31:59.004337+0200	yabai	server port 0x00002e0b, session port 0x00002e0b
default	11:31:59.005215+0200	tccd	AUTHREQ_ATTRIBUTION: msgID=408.525, attribution={responsible={TCCDProcess: identifier=dev.warp.Warp-Stable, pid=1029, auid=501, euid=501, responsible_path=/Applications/Warp.app/Contents/MacOS/stable, binary_path=/Applications/Warp.app/Contents/MacOS/stable}, accessing={TCCDProcess: identifier=com.koekeishiya.yabai, pid=10577, auid=501, euid=0, binary_path=/usr/local/bin/yabai}, requesting={TCCDProcess: identifier=com.apple.WindowServer, pid=408, auid=88, euid=88, binary_path=/System/Library/PrivateFrameworks/SkyLight.framework/Versions/A/Resources/WindowServer}, },
default	11:31:59.005234+0200	tccd	requestor: TCCDProcess: identifier=com.apple.WindowServer, pid=408, auid=88, euid=88, binary_path=/System/Library/PrivateFrameworks/SkyLight.framework/Versions/A/Resources/WindowServer is checking access for accessor TCCDProcess: identifier=com.koekeishiya.yabai, pid=10577, auid=501, euid=0, binary_path=/usr/local/bin/yabai
default	11:31:59.019501+0200	yabai	[0x80542c640] activating connection: mach=true listener=false peer=false name=com.apple.coreservices.launchservicesd
default	11:31:59.019889+0200	yabai	[0x80542c780] activating connection: mach=true listener=false peer=false name=com.apple.analyticsd
default	11:31:59.021619+0200	yabai	New connection 0x264ccf main
default	11:31:59.021758+0200	yabai	Current system appearance, (HLTB: 1), (SLS: 0)
default	11:31:59.022406+0200	yabai	No persisted cache on this platform.
default	11:31:59.025352+0200	yabai	Received configuration update from daemon (initial)
default	11:31:59.025907+0200	yabai	Current system appearance, (HLTB: 1), (SLS: 0)
default	11:31:59.027618+0200	yabai	Post-registration system appearance: (HLTB: 1)
default	11:31:59.030178+0200	yabai	Entering exit handler.
default	11:31:59.030184+0200	yabai	Queueing exit procedure onto XPC queue. Any further messages sent will be discarded. activeSendTransactions=0
default	11:31:59.030205+0200	yabai	Cancelling XPC connection. Any further reply handler invocations will not retry messages
default	11:31:59.030208+0200	yabai	[0x80542c780] invalidated because the current process cancelled the connection by calling xpc_connection_cancel()
error	11:31:59.030596+0200	usernoted	Record <NotificationRecord app:"com.koekeishiya.yabai" ident:"DA39-A3EE" req:"" uuid:"8505B7DD" source:"AB03AB31" staticCategory:"<LEGACY options=(legacyBehavior, hiddenPreviewShowsTitle) actions=[
    <ACTION style=button title=''>
  ]>" <response action: closed actIdent:  foreground: true text: nil> > ineligible for pipeline due to summarizationDisabled(com.koekeishiya.yabai)
default	11:31:59.030215+0200	yabai	Exiting exit handler.
default	11:31:59.030615+0200	usernoted	<NotificationRecord app:"com.koekeishiya.yabai" ident:"DA39-A3EE" req:"" uuid:"8505B7DD" source:"AB03AB31" staticCategory:"<LEGACY options=(legacyBehavior, hiddenPreviewShowsTitle) actions=[
    <ACTION style=button title=''>
  ]>" <response action: closed actIdent:  foreground: true text: nil> > successfully processed by pipeline, scheduled for delivery.
default	11:31:59.135104+0200	usernoted	Delivering <NotificationRecord app:"com.koekeishiya.yabai" ident:"DA39-A3EE" req:"" uuid:"8505B7DD" source:"AB03AB31" staticCategory:"<LEGACY options=(legacyBehavior, hiddenPreviewShowsTitle) actions=[
    <ACTION style=button title=''>
  ]>" <response action: closed actIdent:  foreground: true text: nil> > to [ .alert .lockScreen .notificationCenter ]: updatedExisting: false
default	11:31:59.149868+0200	NotificationCenter	[com.apple.private.nc.dnd:E5DFDC82-5A85-439A-AB5B-54726B23469A] Resolving behavior for event, details=<DNDMutableClientEventDetails: 0x9ad399000; identifier: ''; bundleIdentifier: com.koekeishiya.yabai; platform: unknown; type: Default; urgency: Default; sender: (null); threadIdentifier: (null); filterCritera: (null); notifyAnyway: 0; behavior: Default; forwardingBehavior: (null); title: 127EEDBA286B; subtitle: 6B9C8C514557B45E; body: 4763F41356F37187; shouldAdjustEventBehaviorForMirroring: 0>
default	11:31:59.150468+0200	donotdisturbd	Event was resolved: resolution=<DNDSEventBehaviorResolution: 0xbb600b680; UUID: EC2E182C-88EA-47DE-8830-2670F40B7F11; date: 2025-09-04 09:31:59 +0000; eventBehavior: <DNDClientEventBehavior: 0xbb7f80960; eventDetails: <DNDClientEventDetails: 0xbb669a180; identifier: ''; bundleIdentifier: com.koekeishiya.yabai; platform: unknown; type: Default; urgency: Default; sender: (null); threadIdentifier: (null); filterCritera: (null); notifyAnyway: 0; behavior: Default; forwardingBehavior: (null); title: 127EEDBA286B; subtitle: 127EEDBA286B; body: 4763F41356F37187; shouldAdjustEventBehaviorForMirroring: 0>; interruptionSuppression: none; intelligentBehavior: unused; resolutionReason: disabled; activeModeUUID: (null)>; clientIdentifier: 'com.apple.nc.donotdisturb.user-toggles.preload'; outcome: allowed; reason: disabled>
default	11:31:59.150682+0200	NotificationCenter	[com.apple.private.nc.dnd:E5DFDC82-5A85-439A-AB5B-54726B23469A] Resolved event, details=<DNDMutableClientEventDetails: 0x9ad399000; identifier: ''; bundleIdentifier: com.koekeishiya.yabai; platform: unknown; type: Default; urgency: Default; sender: (null); threadIdentifier: (null); filterCritera: (null); notifyAnyway: 0; behavior: Default; forwardingBehavior: (null); title: 127EEDBA286B; subtitle: 6B9C8C514557B45E; body: 4763F41356F37187; shouldAdjustEventBehaviorForMirroring: 0> behavior=<DNDClientEventBehavior: 0x9a9bed680; eventDetails: <DNDClientEventDetails: 0x9ad39a500; identifier: ''; bundleIdentifier: com.koekeishiya.yabai; platform: unknown; type: Default; urgency: Default; sender: (null); threadIdentifier: (null); filterCritera: (null); notifyAnyway: 0; behavior: Default; forwardingBehavior: (null); title: 127EEDBA286B; subtitle: 127EEDBA286B; body: 4763F41356F37187; shouldAdjustEventBehaviorForMirroring: 0>; interruptionSuppression: none; intelligentBehavior: unused; resolutionReason: disabled; activeModeUUID: (
default	11:31:59.150840+0200	NotificationCenter	record <NotificationRecord app:"com.koekeishiya.yabai" ident:"DA39-A3EE" req:"" uuid:"8505B7DD" source:"AB03AB31" staticCategory:"<LEGACY options=(legacyBehavior, hiddenPreviewShowsTitle) actions=[
    <ACTION style=button title=''>
  ]>" <response action: closed actIdent:  foreground: true text: nil> > is not intelligentlyBrokethrough
default	11:31:59.150908+0200	NotificationCenter	addOrUpdate listItem: com.koekeishiya.yabai:8505B7DD from app com.koekeishiya.yabai, canDisplayWhileCenterIsClosed: false, visibility: [], readCount: 0
default	11:31:59.150921+0200	NotificationCenter	Adding notification to storage: com.koekeishiya.yabai:8505B7DD
```

tbiehn commented:
> This is what I get:
> 
> ```
> default	12:41:14.831926+0200	Dock	[yabai-sa] loaded payload..
> default	12:41:14.832143+0200	Dock	[yabai-sa] checking for macOS 26.0.0 compatibility!
> default	12:41:14.832173+0200	Dock	[yabai-sa] spaces functionality is only supported on macOS Monterey 12.0.0+, and Ventura 13.0.0+, Sonoma 14.0.0+, and Sequoia 15.0
> default	12:41:14.832201+0200	Dock	[yabai-sa] now listening..
> ```
> 
> That looks normal, no? 🤔
> 
> The command `yabai -m space --switch 2` still doesn't actually moves to that space. Maybe I'm overlooking something?

Nope - that's telling me that the SA thats running isn't actually from this fork. If you were you'd see a message related to Tahoe being detected - just take a look at my screenshot.
Double check your binaries and paths and **really** convince yourself you're running off this patch.

roelvangils commented:
@tbiehn

> Double check your binaries and paths and really convince yourself you're running off this patch.

Thanks. I wasn't running Yabai from the patch; I was using the Homebrew version. I've got everything working now. I won't touch it again until @koekeishiya merges your changes. I'm not a developer, so I had to experiment a bit to get it to work. Thanks for your help!

(I can't imagine using spaces anymore with the horrible macOS swipe / fade animation.)

roelvangils commented:
Update: after rebooting, it stopped working again and I can't get it to work anymore.

```
yabai -m space --switch 2
yabai-msg: failed to connect to socket..
```

I removed the socket files in /tmp, deleted Yabai from accessibility permissions, and stopped and restarted the service. I'm not sure what else I can try. Help would be appreciated :)

tbiehn commented:
> Update: after rebooting, it stopped working again and I can't get it to work anymore.
> 
> ```
> yabai -m space --switch 2
> yabai-msg: failed to connect to socket..
> ```
> 
> I removed the socket files in /tmp, deleted Yabai from accessibility permissions, and stopped and restarted the service. I'm not sure what else I can try. Help would be appreciated :)

So console.app is how you debug SA problems.
I’m going to guess that the shasum in sudoers is wrong. So you need to update that so that the load-sa command in yabairc succeeds.
You gotta whitelist the shasum of the binary every single time yabai is updated in sudoers. Not just once on install.
Any other problems you’re gonna have to show me console.app output for yabai-sa.

tbiehn commented:
> @tbiehn I'd be happy to try it out. I have an extra iMac 27" 2020 (Intel i7) on my desk with the latest public beta. If you provide me with step-by-step instructions on how to test it, I'd be glad to help.

If you're feeling bullish now that you've done it once on your M chip - now's the time to run it on the i7!

roelvangils commented:
> If you're feeling bullish now that you've done it once on your M chip - now's the time to run it on the i7!

Ha, okay ;) It's in the office. I'll try it on Monday!

stephen-zeng commented:
@tbiehn RC has been released, can you update the merge? I can't wait to bring yabai back😀

tbiehn commented:
Tested on RC1 - working without changes.

@koekeishiya - Patterns remain stable and unchanged for ARM and x86_64 targets (untested, AFAIK)  in the RC. 

@stephen-zeng - Should be working for you on RC1 25A353 

84d9008762ebf0c5efb4084ec75384fd55504cb6e1e83f0170caa25b3c3be2bf  /System/Library/CoreServices/Dock.app/Contents/MacOS/Dock

koekeishiya commented:
I'll see if I can make time this weekend to look things over for the Tahoe release.

auscyber commented:
i've been running this on nix-darwin for. the past few days (building it with my nur package) and haven't had many problems except having to do a few pkill yabai and uninstalls but right now with the dock restart hook, i havent had any problems

koekeishiya commented:
I don't think the x64 one is fully correct, but I'll just merge this for now. Thanks.
